internal/config/config.go

```diff
@@ -41,21 +47,79 @@ type Config struct {
 	WebhookPort            int            `yaml:"webhookPort"`
 	Worker                 Worker         `yaml:"worker"`
 }
+type ConfigGetter interface {
+	GetConfig(ctx context.Context, logger logr.Logger, userConfigMapName, userConfigMapNamespace string) (*Config, error)
+}
+
+//go:generate mockgen -source=config.go -package=config -destination=mock_config.go
```

Comment:
Let's have a single `generate` annotation before `ConfigGetter` interface definition with both interfaces in it: like we do at https://github.com/kubernetes-sigs/kernel-module-management/blob/925409699c648c0eb747b34bc7b2e06b425a1c23/internal/module/kernelmapper.go#L16